core/engine_test.go

```diff
@@ -691,7 +691,7 @@ func TestRunTags(t *testing.T) {
 	tb := httpmultibin.NewHTTPMultiBin(t)
 
 	runTagsMap := map[string]string{"foo": "bar", "test": "mest", "over": "written"}
-	runTags := metrics.NewSampleTags(runTagsMap)
+	runTags := metrics.NewSampleTags(runTagsMap).CloneTags()
```

Comment:
why is this done? won't `runTagsMap` be exactly equivalent to `runTags`?

Reply:
In the case, there is a bug and the runner overwrites the run tags then the map would be changed and the assertion will pass, but it shouldn't.

I have updated it a bit the code and added a comment, so it would be clearer.